onnxruntime/test/onnx/main.cc

```diff
@@ -786,12 +838,62 @@ select from 'TF8', 'TF16', 'UINT8', 'FLOAT', 'ITENSOR'. \n)");
+    {"castlike_BFLOAT16_to_FLOAT_expanded", "Unexpected input data type"},
     {"castlike_FLOAT_to_STRING", "Numpy float to string has unexpected rounding for some results.", {}},
     {"castlike_FLOAT_to_STRING_expanded", "Numpy float to string has unexpected rounding for some results.", {}},
+    {"cast_FLOAT8E5M2_to_FLOAT", "Unexpected input data type"},
+    {"cast_FLOAT8E5M2_to_FLOAT16", "Unexpected input data type"},
+    {"cast_FLOAT_to_FLOAT8E4M3FN", "Unexpected input data type"},
+    {"cast_FLOAT_to_FLOAT8E4M3FNUZ", "Unexpected input data type"},
+    {"cast_FLOAT_to_FLOAT8E5M2", "Unexpected input data type"},
+    {"cast_FLOAT_to_FLOAT8E5M2FNUZ", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT16_to_FLOAT8E4M3FN", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT16_to_FLOAT8E4M3FNUZ", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT16_to_FLOAT8E5M2", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT16_to_FLOAT8E5M2FNUZ", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT_to_FLOAT8E4M3FN", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT_to_FLOAT8E4M3FNUZ", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT_to_FLOAT8E5M2", "Unexpected input data type"},
+    {"cast_no_saturate_FLOAT_to_FLOAT8E5M2FNUZ", "Unexpected input data type"},
+    {"castlike_FLOAT8E4M3FNUZ_to_FLOAT", "Unexpected input data type"},
+    {"castlike_FLOAT8E4M3FNUZ_to_FLOAT_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT8E4M3FN_to_FLOAT", "Unexpected input data type"},
+    {"castlike_FLOAT8E4M3FN_to_FLOAT_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT8E5M2FNUZ_to_FLOAT", "Unexpected input data type"},
+    {"castlike_FLOAT8E5M2FNUZ_to_FLOAT_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT8E5M2_to_FLOAT", "Unexpected input data type"},
+    {"castlike_FLOAT8E5M2_to_FLOAT_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E4M3FN", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E4M3FNUZ", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E4M3FNUZ_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E4M3FN_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E5M2", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E5M2FNUZ", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E5M2FNUZ_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT_to_FLOAT8E5M2_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT_to_STRING_expanded", "Unexpected input data type"},
+    {"castlike_FLOAT_to_STRING", "Unexpected input data type"},
+    {"castlike_FLOAT_to_BFLOAT16", "Unexpected input data type"},
+    {"castlike_BFLOAT16_to_FLOAT_expanded", "Unexpected input data type"},
+    {"castlike_BFLOAT16_to_FLOAT", "Unexpected input data type"},
+    {"cast_FLOAT16_to_FLOAT8E4M3FNUZ", "Unexpected input data type"},
```

Comment:
adding "cast_FLOAT16_to_FLOAT8E4M3FN" here should fix the build failure.

then, I think we can revert the disabled/broken test list updates in the other two places as those are for when QNN EP is enabled.

the reason string used elsewhere, "float8 is not implemented yet", may be more meaningful.

Reply:
Thanks I missed it.

Reply:
`Unexpected input data type` is the displayed error message. I did not know if there was a rule about the reason. I can switch.